canopen/objectdictionary/xdd.py

```diff
@@ -0,0 +1,306 @@
+import logging
+
+import re
+import xml.etree.ElementTree as etree
+from configparser import NoOptionError
+from typing import TYPE_CHECKING
+
+from canopen import objectdictionary
+from canopen.objectdictionary import ObjectDictionary
+from canopen.utils import signed_int_from_hex, calc_bit_length
+
+if TYPE_CHECKING:
+    import canopen.network
+
+logger = logging.getLogger(__name__)
+
+# Object type. Don't confuse with Data type
+VAR = 7
+ARR = 8
+RECORD = 9
+
+
+def import_xdd(xdd, node_id):
+    od = ObjectDictionary()
+    if etree.iselement(xdd):
+        root = xdd
+    else:
+        root = etree.parse(xdd).getroot()
+
+    if node_id is None:
+        device_commissioning = root.find('.//{*}DeviceCommissioning')
+        if device_commissioning is not None:
+            if node_id := device_commissioning.get('nodeID', None):
+                try:
+                    od.node_id = int(node_id, 0)
+                except (ValueError, TypeError):
+                    pass
+    else:
+        od.node_id = node_id
+
+    _add_device_information_to_od(od, root)
+    _add_object_list_to_od(od, root)
+    _add_dummy_objects_to_od(od, root)
+
+
+    return od
+
+def _add_device_information_to_od(od, root):
+    device_identity = root.find('.//{*}DeviceIdentity')
+    if device_identity is not None:
+        for src_prop, dst_prop, f in [
+            ("vendorName", "vendor_name", lambda val: str(val)),
```

Comment:
```suggestion
            ("vendorName", "vendor_name", str),
```
This can be simplified in many places, no need for a lambda.